src/Chainweb/Chainweb.hs

```diff
@@ -444,78 +443,95 @@ runChainweb
-
-        payloadDbsToServe = itoList $ const (view chainwebPayloadDb cw) <$> _chainwebChains cw
-        pactDbsToServe = _chainwebPactData cw
-
-        serverSettings
-            = setOnException
-                (\r e -> when (defaultShouldDisplayException e) (logg Error $ loggServerError r e))
-            $ peerServerSettings (_peerResPeer $ _chainwebPeer cw)
-        serve = serveChainwebSocketTls
-            serverSettings
-            (_peerCertificateChain $ _peerResPeer $ _chainwebPeer cw)
-            (_peerKey $ _peerResPeer $ _chainwebPeer cw)
-            (_peerResSocket $ _chainwebPeer cw)
-            (_chainwebVersion cw)
-            ChainwebServerDbs
-                { _chainwebServerCutDb = Just cutDb
-                , _chainwebServerBlockHeaderDbs = chainDbsToServe
-                , _chainwebServerMempools = mempoolsToServe
-                , _chainwebServerPayloadDbs = payloadDbsToServe
-                , _chainwebServerPeerDbs = (CutNetwork, cutPeerDb) : chainP2pToServe
-                , _chainwebServerPactDbs = pactDbsToServe
-                }
-
-        -- HTTP Request Logger
-        httpLog :: Middleware
-        httpLog = requestResponseLogger $ setComponent "http" (_chainwebLogger cw)
-
-    -- 1. start server
     --
     withAsync (serve $ throttle (_chainwebThrottler cw) . httpLog) $ \server -> do
         logg Info "started server"
 
-        -- Configure Clients
+        -- 2. Start Clients
         --
-        let mgr = view chainwebManager cw
-            miner = maybe [] (\m -> [ runMiner (_chainwebVersion cw) m ]) $ _chainwebMiner cw
-            -- run mempool sync without manager timeout; servant default is ok
-            -- for non-streaming endpoints and mempool does its own timeout
-            -- bookkeeping for the streaming endpoints
-            mempoolMgr = mgr { HTTP.mResponseTimeout = HTTP.responseTimeoutNone }
-
-        -- 2. Run Clients
-        --
-        let clients :: [IO ()]
-            clients = concat
-                [ miner
-                , cutNetworks mgr (_chainwebCutResources cw)
-                , map (runMempoolSyncClient mempoolMgr mempoolP2pConfig) chainVals
-                ]
-
         mapConcurrently_ id clients
         wait server
+
   where
     logg = logFunctionText $ _chainwebLogger cw
-    mempoolP2pConfig = _configMempoolP2p $ _chainwebConfig cw
+
+    -- chains
+    chains = HM.toList (_chainwebChains cw)
+    chainVals = map snd chains
+
+    -- collect server resources
+    proj :: forall a . (ChainResources logger -> a) -> [(ChainId, a)]
+    proj f = flip map chains $ \(k, ch) -> (k, f ch)
+
+    chainDbsToServe = proj _chainResBlockHeaderDb
+    mempoolsToServe = proj _chainResMempool
+    chainP2pToServe = bimap ChainNetwork (_peerResDb . _chainResPeer) <$> itoList (_chainwebChains cw)
+    memP2pToServe = bimap MempoolNetwork (_peerResDb . _chainResPeer) <$> itoList (_chainwebChains cw)
```

Comment:
Defining `memP2pToServe` and using it below in line 491 is the fix.

All the other changes in this PR are just cleanup of the Chainweb initialization.